ruby_event_store/lib/ruby_event_store/projection.rb

```diff
@@ -85,7 +85,7 @@ def start_events(start)
     end
 
     def transition(state, event)
-      handlers[event.class].(state, event)
+      handlers[event.type].(state, event)
```

Comment:
why no `fetch` here?

Reply:
Good question - but should it be? or maybe we want to ignore events for which there is no defined handlers?

Currently when you do not specify any stream to process projection works on all events in all streams (`reduce_from_all_streams` method). Current implementation forces you to provide handlers for all domain event types :( Good luck on some production systems with reasonable number of domain events ;)

Reply:
```
      @handlers = Hash.new { ->(_, _) {} }
```

That's why it's not fetch. I was wrong ... no need to define all kind of events - I've missed that.
`handlers[event.type]` for not existing event type will return "empty" lambda.